src/reducers/vms.js

```diff
@@ -77,11 +79,24 @@ const vms = actionReducer(initialState, {
     if (page) {
       st = st.set('page', page)
     }
+
+    const vmsIds = Object.keys(updates)
+    const filteredSet = new Set()
+    const missedVms = st.get('missedVms')
+    for (const vmId of vmsIds) {
+      if (!missedVms.has(vmId)) {
+        filteredSet.add(vmId)
+      }
+    }
+
+    st = st.set('missedVms', filteredSet)
```

Comment:
Much easier if you utilize ImmutableJS APIs:
```suggestion
    st = st.set('missedVms', st.get('missedVms').subtract(vms.map(vm => vm.id)))
```